crates/ruff_python_formatter/src/other/comprehension.rs

```diff
@@ -1,12 +1,115 @@
 
 impl FormatNodeRule<Comprehension> for FormatComprehension {
     fn fmt_fields(&self, item: &Comprehension, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let Comprehension {
+            range: _,
+            target,
+            iter,
+            ifs,
+            is_async,
+        } = item;
+
+        let comments = f.context().comments().clone();
+        let leading_item_comments = comments.leading_comments(item);
+
+        // TODO: why is this needed?
+        if let Some(leading_item_comment) = leading_item_comments.first() {
+            if leading_item_comment.line_position().is_own_line() {
+                hard_line_break().fmt(f)?;
+            }
+        }
+        leading_comments(leading_item_comments).fmt(f)?;
+
+        if *is_async {
+            write!(f, [text("async"), space()])?;
+        }
+
+        let dangling_item_comments = comments.dangling_comments(item);
+
+        let (before_target_comments, before_in_comments) = dangling_item_comments.split_at(
+            dangling_item_comments
+                .partition_point(|comment| comment.slice().end() < target.range().start()),
+        );
+
+        let trailing_in_comments = comments.dangling_comments(iter);
+        write!(
+            f,
+            [
+                text("for"),
+                trailing_comments(before_target_comments),
+                group(&self::format_args!(
+                    soft_line_break_or_space(),
+                    &target.format(),
+                    soft_line_break_or_space(),
+                    leading_comments(before_in_comments),
+                    text("in"),
+                    trailing_comments(trailing_in_comments),
+                    soft_line_break_or_space(),
+                    iter.format(),
+                )),
+            ]
+        )?;
+        if !ifs.is_empty() {
+            let joined = format_with(|f| {
+                let mut joiner = f.join_with(soft_line_break_or_space());
+                for if_ in ifs {
+                    let dangling_if_comments = comments.dangling_comments(if_);
+
+                    let (own_line_if_comments, end_of_line_if_comments) = dangling_if_comments
+                        .split_at(
+                            dangling_if_comments
+                                .partition_point(|comment| comment.line_position().is_own_line()),
+                        );
+                    joiner.entry(&group(&self::format_args!(
+                        leading_comments(own_line_if_comments),
+                        text("if"),
+                        trailing_comments(end_of_line_if_comments),
+                        soft_line_break_or_space(),
+                        if_.format(),
+                    )));
+                }
+                joiner.finish()
+            });
+
+            write!(f, [soft_line_break_or_space(), group(&joined)])?;
+        }
+        Ok(())
+    }
+
```

Comment:
We can override `fmt_dangling_comments` here since you're handling it in the `fmt_fields` function